core/src/main/java/io/neow3j/protocol/rx/JsonRpc2_0Rx.java

```diff
@@ -31,32 +28,50 @@ public JsonRpc2_0Rx(Neow3j neow3j, ScheduledExecutorService scheduledExecutorSer
         this.scheduler = Schedulers.from(scheduledExecutorService);
     }
 
-    public Observable<BigInteger> neoBlockObservable(long pollingInterval) {
-        return Observable.create(subscriber -> {
-            BlockPolling blockPolling = new BlockPolling(neow3j, subscriber::onNext);
-            blockPolling.run(scheduledExecutorService, pollingInterval);
-            subscriber.setDisposable(Disposables.fromAction(blockPolling::cancel));
-        });
+    /**
+     * Creates an observable that emits new block index as they are produced by the Neo blockchain. The observable
+     * polls the Neo node in the given {@code pollingInterval} to check for the latest block index and emits all
+     * indexes since the last time it polled.
+     *
+     * @param pollingInterval The polling interval in milliseconds.
+     * @return the block index observable.
+     */
+    public Observable<BigInteger> blockIndexObservable(long pollingInterval) {
+        return Observable.create(subscriber ->
+                new BlockIndexPolling().run(neow3j, subscriber, scheduledExecutorService, pollingInterval)
+        );
     }
 
-    public Observable<NeoGetBlock> replayBlocksObservable(BigInteger startBlock, BigInteger endBlock,
-            boolean fullTransactionObjects) {
-        return replayBlocksObservable(startBlock, endBlock, fullTransactionObjects, true);
```

Comment:
I removed this without side effects just because I wanted to strip down this class to the bear minimum for better maintainability. Whatever neow3j code used this method now simply uses the corresponding overload with an extra boolean argument at the end. Let me know if you don't agree at all with such cleanups.